src/Interpreters/AsynchronousInsertQueue.h

```diff
@@ -105,6 +108,7 @@ class AsynchronousInsertQueue : public WithContext
     /// (async_insert_max_data_size setting). If so, then again we dump the data.
 
     std::atomic<bool> shutdown{false};
+    std::atomic<bool> flushes_cancelled{false};
```

Comment:
Just a suggestion: "flushes_enabled" or "flushes_disabled" seems to be a better name